src/util.rs

```diff
@@ -24,6 +25,41 @@ use windows::{
     },
 };
 
+pub(crate) const fn win_guid_to_u128(guid: &GUID) -> u128 {
+    let data4_u64 = u64::from_be_bytes(guid.data4);
+    ((guid.data1 as u128) << 96) | ((guid.data2 as u128) << 80) | ((guid.data3 as u128) << 64) | (data4_u64 as u128)
+}
+
+pub(crate) fn win_pstr_to_string(pstr: ::windows_sys::core::PSTR) -> Result<String, Error> {
```

Comment:
This needs to be marked unsafe as otherwise it is possible to break memory safety in safe code. https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html#method.from_ptr is a good example of what I'm thinking since were basically doing the same thing